harper-core/src/linting/closed_compounds.rs

```diff
@@ -0,0 +1,342 @@
+    Token, TokenStringExt,
+};
+
+use super::{Lint, LintKind, PatternLinter, Suggestion};
+
+macro_rules! create_closed_compound_linter {
+    ($name:ident, $phrase:literal, $correct:expr) => {
+        pub struct $name {
+            pattern: Box<dyn Pattern>,
+        }
+
+        impl Default for $name {
+            fn default() -> Self {
+                let pattern = ExactPhrase::from_phrase($phrase);
+
+                Self {
+                    pattern: Box::new(pattern),
+                }
+            }
+        }
+
+        impl PatternLinter for $name {
+            fn pattern(&self) -> &dyn Pattern {
+                self.pattern.as_ref()
+            }
+
+            fn match_to_lint(&self, matched_tokens: &[Token], source: &[char]) -> Lint {
+                let span = matched_tokens.span().unwrap();
+                let orig_chars = span.get_content(source);
+
+                Lint {
+                    span,
+                    lint_kind: LintKind::WordChoice,
+                    suggestions: vec![Suggestion::replace_with_match_case(
+                        $correct.chars().collect(),
+                        orig_chars,
+                    )],
+                    message: format!("Did you mean the closed compound `{}`?", $correct),
+                    ..Default::default()
+                }
+            }
+
+            fn description(&self) -> &'static str {
+                concat!(
+                    "Looks for incorrect spacing inside the closed compound `",
+                    $correct,
+                    "`."
+                )
+            }
+        }
+    };
+}
+
+create_closed_compound_linter!(Itself, "it self", "itself");
+create_closed_compound_linter!(Tonight, "to night", "tonight");
+create_closed_compound_linter!(Myself, "my self", "myself");
+create_closed_compound_linter!(Therefore, "there fore", "therefore");
+create_closed_compound_linter!(Misunderstand, "miss understand", "misunderstand");
+create_closed_compound_linter!(Misunderstood, "miss understood", "misunderstood");
+create_closed_compound_linter!(Misuse, "miss use", "misuse");
+create_closed_compound_linter!(Misused, "miss used", "misused");
+create_closed_compound_linter!(Postpone, "post pone", "postpone");
+create_closed_compound_linter!(Worldwide, "world wide", "worldwide");
+create_closed_compound_linter!(Overall, "over all", "overall");
+create_closed_compound_linter!(Likewise, "like wise", "likewise");
+create_closed_compound_linter!(However, "how ever", "however");
+create_closed_compound_linter!(Altogether, "all together", "altogether");
+create_closed_compound_linter!(Asleep, "a sleep", "asleep");
+create_closed_compound_linter!(Upset, "up set", "upset");
+create_closed_compound_linter!(Intact, "in tact", "intact");
+create_closed_compound_linter!(Somehow, "some how", "somehow");
+create_closed_compound_linter!(Proofread, "proof read", "proofread");
+create_closed_compound_linter!(Nobody, "no body", "nobody");
```

Comment:
Both are legit. Your hair can have no body, a character or creature can be just a head with no body. Browser can distinguish: 
<img width="357" alt="image" src="https://github.com/user-attachments/assets/0eed1fab-9825-444b-b074-bd01ed722e12" />